### DISCUSSION TITLE
Disable DNS Leak feature on OpenVPN.conf

### DISCUSSION THREAD

**Discussion by EliotLeBon** [qdm12/gluetun#1531] [Q&A]
Hi,

I use Gluetun with a custom ovpn config and i'd like to disable the anti-leak dns feature.

When i start my container, the logs output this : 
`gluetun       | 2023-04-19T10:59:07+02:00 ERROR VPN settings: OpenVPN settings: custom configuration file: extracting information from custom configuration file: extracting connection from file: on line 4: extracting from remote line: host is not an an IP address: fr-bod.prod.surfshark.com
gluetun       | 2023-04-19T10:59:07+02:00 INFO Shutdown successful
`

I can read on this page :  https://github.com/qdm12/gluetun/wiki/OpenVPN-configuration-file
`Replace the VPN server hostname by one of its IP addresses. In custom.conf, find the line starting with remote. The second field is the VPN server hostname. If it is not an IP address, you need to DNS resolve it, for example with nslookup domain.com and replace the hostname field with one of its corresponding IP addresses. This is the case as gluetun's firewall is designed not to leak anything including an initial DNS resolution when starting.`

I've trying to disable it with FIREWALL=off / DNS_KEEP_NAMESERVER=on / BLOCK_MALICIOUS=off but it doesnt work.

Do you have any idea ?

Eliot :)

**Comment by bnhf** [comment]
@EliotFR44 

I'm not aware of any way to do what you're asking with Gluetun. Preventing DNS leaks is baked into this project, and one of its many strengths.

Here's what I got resolving ```fr-bod.prod.surfshark.com``` into its IP address(es):

```console
Name:    fr-bod.prod.surfshark.com
Addresses:  45.134.79.163
            45.134.79.153
```

**Reply by EliotLeBon** [reply]
Yes but it changes all the time :)
![ip](https://user-images.githubusercontent.com/39886697/233128980-5153f207-0cda-4efe-9947-951262f5a8ae.PNG)

I don't want to update them all the time.

**Reply by bnhf** [reply]
@EliotFR44 

> Yes but it changes all the time :)

Looks like they have quite a few IPs for this particular hostname. I tried pinging each of the responses you got, and they all answered. Two of your responses were the same as what I got.

Also, doing half-a-dozen more lookups, I saw most addresses repeat. Seems like just a long list to me, where you could either just pick one, or you should be able to expand your .ovpn file to include a number of them, each on their own line:

```console
remote 45.134.79.131 1194 udp # Modify the port and protocol as appropriate
remote 45.134.79.136 1194 udp
remote 45.134.79.138 1194 udp
remote 45.134.79.141 1194 udp
remote 45.134.79.151 1194 udp
remote 45.134.79.153 1194 udp
remote 45.134.79.158 1194 udp
remote 45.134.79.163 1194 udp
```

Looking at servers.json, this hostname is listed, but it appears the addresses are no longer valid -- which I imagine is why you're trying to do it through a custom .ovpn:

```json
     {
        "vpn": "openvpn",
        "country": "France",
        "region": "Europe",
        "city": "Bordeaux",
        "hostname": "fr-bod.prod.surfshark.com",
        "tcp": true,
        "udp": true,
        "retroloc": "France Bordeaux",
        "ips": [
          "185.108.106.74",
          "185.108.106.106",
          "185.108.106.144",
          "185.108.106.156",
          "185.108.106.160",
          "185.108.106.164",
          "185.108.106.171",
          "185.108.106.184"
        ]
      },

```

**Reply by EliotLeBon** [reply]
Actually, i dev a program who balance somes requests on the right tunnel.
I already manage if a tunnel is down, blacklisted, etc ....

i use the custom method because i don't want gluetun choose for me where does my trafic out.

So, i've bruteforce the ip range of fr-bord and voila : 
![fr-bord](https://user-images.githubusercontent.com/39886697/233170369-e42f8801-d6d5-4f92-b5e6-47fdb724ef83.PNG)
